std/range/package.d

```diff
@@ -9761,10 +9767,13 @@ private struct OnlyResult(T, size_t arity)
         ++frontIndex;
     }
 
-    T back() @property
+    CommonType!Values back() @property
     {
         assert(!empty, "Attempting to fetch the back of an empty Only range");
-        return data[backIndex - 1];
+        final switch (backIndex - 1)
+            static foreach (i, T; Values)
+            case i:
+                return values[i];
```

Comment:
Can this be combined into one function with front, back, and opIndex?

Reply:
done, switched to `this[0]` / `this[$ - 1]`.